docs/AdvancedContributorGuidelines.md

```diff
@@ -0,0 +1,49 @@
+# Advanced Contributor Guidelines
+
+## Choosing issues
+
+* Follow issue priorities. Prefer to take higher priority issues over lower priority issues.
+* Favor issues related to certain features/aspects/techs to deepen expertise in those areas. 
+ 
+## Creating commits in big PRs
```

Comment:
@pyokagan I agree. We should make it a hard requirement. 

It's hard to define 'big' even in LoC because sometimes a trivial change (an invasive refactoring) can touch a lot of lines. I think what we are looking for is PRs that are too big to review in one-shot (i.e. a 15 minute window); needs to be reviewed incrementally instead.

Perhaps it's the reviewer, realizing the PR is too big, who should ask the dev to refactor the commits to make it a `multi-step PR`? I also suspect, given devs are not very experienced, only in retrospect they can figure out the correct sequence of steps. So creating commits to reflect the steps will be done after the whole change is done anyway. Senior members (PM, project lead, etc.) who monitor incoming PRs can also step in to intercept big PRs and ask the dev to update it to multi-step format.